packages/block-editor/src/hooks/layout-child.js

```diff
@@ -15,10 +15,23 @@ function useBlockPropsChildLayoutStyles( { style } ) {
 		return ! select( blockEditorStore ).getSettings().disableLayoutStyles;
 	} );
 	const layout = style?.layout ?? {};
-	const { selfStretch, flexSize } = layout;
+	const { selfStretch, flexSize, columnSpan, rowSpan, parentColumnWidth } =
+		layout;
 	const id = useInstanceId( useBlockPropsChildLayoutStyles );
 	const selector = `.wp-container-content-${ id }`;
 
+	// Calculate container query if needed.
+	const columnSpanNumber = parseInt( columnSpan );
+	const parentColumnValue = parseFloat( parentColumnWidth );
+	const parentColumnUnit = parentColumnWidth?.replace(
+		parentColumnValue,
+		''
+	);
+	const defaultGapValue = parentColumnUnit === 'px' ? 24 : 1.5;
+	const containerQueryValue =
+		columnSpanNumber * parentColumnValue +
+		( columnSpanNumber - 1 ) * defaultGapValue;
+
```

Comment:
Not a blocker, but a tiny optimisation for the future: if `style?.layout` is empty, then we probably don't need to do any calculations 🤔

Reply:
I'll move these calculations into the `columnSpan && parentColumnWidth `  condition where they're relevant. Unfortunately we're not able to return early because of the `useStyleOverride` hook below.